src/app/rundown-view/components/rundown/rundown.component.ts

```diff
@@ -72,17 +76,54 @@ export class RundownComponent implements OnInit, OnDestroy, OnChanges {
   }
 
   private subscribeToEventObserver(): void {
-    this.rundownEventSubscriptions.push(this.rundownEventObserver.subscribeToRundownTake(event => this.scrollViewToSegment(event)))
-    this.rundownEventSubscriptions.push(this.rundownEventObserver.subscribeToRundownSetNext(event => this.scrollViewToSegment(event)))
+    this.rundownEventSubscriptions.push(this.rundownEventObserver.subscribeToRundownTake(event => this.prioritizedDebounce(event)))
+    this.rundownEventSubscriptions.push(this.rundownEventObserver.subscribeToRundownSetNext(event => this.prioritizedDebounce(event)))
   }
 
-  public scrollViewToSegment(event: PartEvent): void {
-    const element: HTMLElement | null = document.getElementById(event.segmentId)
-    if (!element || this.isInsideViewport(element)) {
+  private prioritizedDebounce(event: PartEvent): void {
+    this.startDebounceTimer()
+    const currentPriority: number = this.highestPriorityEvent ? this.getEventPriority(this.highestPriorityEvent) : 0
+    const eventPriority: number = this.getEventPriority(event)
+    if (eventPriority >= currentPriority) {
+      this.highestPriorityEvent = event
+    }
+  }
+
+  private startDebounceTimer(): void {
+    this.clearDebounceTimer()
+    this.debounceTimer = setTimeout(() => {
+      this.clearDebounceTimer()
+      if (!this.highestPriorityEvent) {
+        return
+      }
+      this.scrollViewToSegment(this.highestPriorityEvent.segmentId)
+      this.highestPriorityEvent = undefined
+    }, DEBOUNCE_EVENT_THRESHOLD_IN_MS)
+  }
+
+  private clearDebounceTimer(): void {
+    clearTimeout(this.debounceTimer)
+    this.debounceTimer = undefined
+  }
+
+  private getEventPriority(event: TypedEvent): number {
+    switch (event.type) {
+      case RundownEventType.TAKEN:
+        return 5
+      case RundownEventType.SET_NEXT:
+        return 4
```

Comment:
I have rearranged the priority, it seemingly fixes the cases that we need due to the margin still showing os the on airs. The only case we don't seem to cover is if we do the exact opposite case of this video, which seems highly unlikely. But before i merge i will play around with an "Adjustment" method, but i won't dedicate too much time on this since our cases are covered for now.